g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/conf/CollectSourceValidationInterceptor.java

```diff
@@ -181,11 +182,24 @@ private static void validateLocale(HttpServletRequest request, List<String> sour
 		}
 		if (!RegExpValidatorUtils.IsLetterAndNumberAndValidchar(locale)) {
 			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
-		}else if (!sourceLocales2.contains(locale)) {
-			throw new VIPAPIException(ValidationMsg.LOCALE_NOT_VALIDE);
+		}
+		if (!sourceLocales2.get(0).equals(ConstantsKeys.FALSE) && !sourceLocales2.contains(locale)) {
+			throw new VIPAPIException(String.format(ValidationMsg.LOCALENAME_ONLY_SUPPORTED, getSourceSupportLocales(sourceLocales2)));
 		}
 	}
 
+	private static String getSourceSupportLocales(List<String> sourceLocaleList) {
+		if(sourceLocaleList == null || sourceLocaleList.size() == 0) {
+			return "";
+		}
+		StringBuilder sb = new StringBuilder();
```

Comment:
Maybe we can use `StringUtils.join` directly, such as
`return StringUtils.join(sourceLocaleList, ConstantsChar.COMMA + ConstantsChar.SPACING);`